src/invidious/videos/parser.cr

```diff
@@ -235,7 +237,14 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
     .dig?("secondaryResults", "secondaryResults", "results")
   secondary_results.try &.as_a.each do |element|
     if item = element["compactVideoRenderer"]?
-      related_video = parse_related_video(item)
+      if rv_published_time_text = item["publishedTimeText"]?
```

Comment:
if `publishedTimeText` is part of the `compactVideoRenderer`, this should be handled inside `parse_related_video`. Overall, variables should be kept in they function and not passed around as much as possible!

Reply:
Sorry for the delay. It's solved in 68611482906f3b70e9c105fec5c57436c877a3b0